packages/kbn-apm-synthtrace/src/cli/utils/start_live_data_upload.ts

```diff
@@ -59,22 +51,46 @@ export async function startLiveDataUpload({
         `Requesting ${new Date(bucketFrom).toISOString()} to ${new Date(bucketTo).toISOString()}`
       );
 
-      const next = logger.perf('execute_scenario', () =>
-        generate({ range: timerange(bucketFrom.getTime(), bucketTo.getTime()) })
-      );
+      const generatorsAndClients = generate({
+        range: timerange(bucketFrom.getTime(), bucketTo.getTime()),
+        client: { logsEsClient, apmEsClient },
+      });
+
+      const generatorsAndClientsArray = castArray(generatorsAndClients);
+
+      const streams = generatorsAndClientsArray.map(({ client }) => {
+        const stream = new PassThrough({ objectMode: true });
+        client.index(stream);
+        return stream;
+      });
 
-      const concatenatedStream = castArray(next)
-        .reverse()
-        .reduce<Writable>((prev, current) => {
-          const currentStream = isGeneratorObject(current) ? Readable.from(current) : current;
-          return currentStream.pipe(prev);
-        }, new PassThrough({ objectMode: true }));
+      process.on('SIGINT', () => closeStreams(streams));
+      process.on('SIGTERM', () => closeStreams(streams));
+      process.on('SIGQUIT', () => closeStreams(streams));
```

Comment:
are these added for each batch? sounds like a memory leak 😛

Reply:
Yeah good catch, i knew i was doing something wrong here